### ISSUE TITLE
Altair Chart displays details out of date range

### ISSUE THREAD

**Issue by e-bengeorge** [altair-viz/altair#2227]
I have  dataset with Date field that ranges from 2020-01-22 to 2020-06-30 and when I construct a line chart it is displaying the line chart for the whole year.

The following is the code I used
```python
alt.Chart(df).mark_line().encode(
    x='monthdate(Date):T',
    y='New cases:Q'
).properties(
    width=500
)
```
How to get the chart only within the date range mentioned.
Thanks in `advance`
![chart](https://user-images.githubusercontent.com/12809525/86326059-2f973400-bc52-11ea-8f25-6262c9aca108.png)
``

**Comment by jakevdp** [comment]
Without more information, it will be difficult to answer your question. Please include a [Minimal Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) that demonstrates the problem you're seeing. For example, here is a minimal reproducible example that matches your description of the data, and does not display the issue you report:
```
import altair as alt
import pandas as pd
import numpy as np

df = pd.DataFrame({
  'Date': pd.date_range(start='2020-01-22', end='2020-06-30', freq='D'),
  'New cases': 100 * np.random.rand(161)
})

alt.Chart(df).mark_line().encode(
    x='monthdate(Date):T',
    y='New cases:Q'
)
```
![visualization - 2020-07-02T045257 717](https://user-images.githubusercontent.com/781659/86355683-e928e180-bc1f-11ea-8276-b489312a22a5.png)

Can you modify this example to demonstrate the issue you're seeing?

**Comment by e-bengeorge** [comment]
It was due to the parsing of date field using pandas. 
Whenever  I edit the dataset(csv format) in WPS office to remove unwanted columns and rename columns and after that if I load the dataset the date and months are messed up.
If I load the dataset directly using pandas and do the cleaning process as dataframe then the date field is correct.
I solved it by using date_parse option to read the date field as" "%d-%m-%Y" format.
It is not the issue with Altair

Thanks Mr.Jake
